### ISSUE TITLE
Multiple definition of 'k2c_linear' and others when compiling test

### ISSUE THREAD

**Issue by rmcconke** [f0uriest/keras2c#6]
I am trying to compile the example test suite. I followed the installation instructions, and I can convert model.h5 (in the example folder) to example.c and example.h:
```
All checks passed
Gathering Weights
Writing layer  input_1:0
Writing layer  dense_1
Writing layer  dense_2
Writing layer  lstm_1
Writing layer  dense_3
Formatted  /home/ryley/WDK/ML/foam/keras2c/example.h
Formatted  /home/ryley/WDK/ML/foam/keras2c/example.c
Writing tests
/home/ryley/anaconda3/envs/keras2c/lib/python3.9/site-packages/keras/engine/training.py:2470: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.
  warnings.warn('`Model.state_updates` will be removed in a future version. '
Formatted  /home/ryley/WDK/ML/foam/keras2c/example_test_suite.c
Done 
C code is in 'example.c' with header file 'example.h' 
Tests are in 'example_test_suite.c' 
```

compiling with the following command

`gcc -std=c99 -I./include/ -o example_test example.c example_test_suite.c -L./include/ -l:libkeras2c.a -lm`

Gives the errors:
```
/usr/bin/ld: /tmp/ccypuWyQ.o:(.bss+0x0): multiple definition of `k2c_linear'; /tmp/ccR5zelS.o:(.bss+0x0): first defined here
/usr/bin/ld: /tmp/ccypuWyQ.o:(.bss+0x8): multiple definition of `k2c_exponential'; /tmp/ccR5zelS.o:(.bss+0x8): first defined here
/usr/bin/ld: /tmp/ccypuWyQ.o:(.bss+0x10): multiple definition of `k2c_relu'; /tmp/ccR5zelS.o:(.bss+0x10): first defined here
/usr/bin/ld: /tmp/ccypuWyQ.o:(.bss+0x18): multiple definition of `k2c_hard_sigmoid'; /tmp/ccR5zelS.o:(.bss+0x18): first defined here
/usr/bin/ld: /tmp/ccypuWyQ.o:(.bss+0x20): multiple definition of `k2c_tanh'; /tmp/ccR5zelS.o:(.bss+0x20): first defined here
/usr/bin/ld: /tmp/ccypuWyQ.o:(.bss+0x28): multiple definition of `k2c_sigmoid'; /tmp/ccR5zelS.o:(.bss+0x28): first defined here
/usr/bin/ld: /tmp/ccypuWyQ.o:(.bss+0x30): multiple definition of `k2c_softmax'; /tmp/ccR5zelS.o:(.bss+0x30): first defined here
/usr/bin/ld: /tmp/ccypuWyQ.o:(.bss+0x38): multiple definition of `k2c_softplus'; /tmp/ccR5zelS.o:(.bss+0x38): first defined here
/usr/bin/ld: /tmp/ccypuWyQ.o:(.bss+0x40): multiple definition of `k2c_softsign'; /tmp/ccR5zelS.o:(.bss+0x40): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_core_layers.o):(.bss+0x0): multiple definition of `k2c_softsign'; /tmp/ccR5zelS.o:(.bss+0x40): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_core_layers.o):(.bss+0x8): multiple definition of `k2c_softplus'; /tmp/ccR5zelS.o:(.bss+0x38): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_core_layers.o):(.bss+0x10): multiple definition of `k2c_softmax'; /tmp/ccR5zelS.o:(.bss+0x30): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_core_layers.o):(.bss+0x18): multiple definition of `k2c_sigmoid'; /tmp/ccR5zelS.o:(.bss+0x28): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_core_layers.o):(.bss+0x20): multiple definition of `k2c_tanh'; /tmp/ccR5zelS.o:(.bss+0x20): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_core_layers.o):(.bss+0x28): multiple definition of `k2c_hard_sigmoid'; /tmp/ccR5zelS.o:(.bss+0x18): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_core_layers.o):(.bss+0x30): multiple definition of `k2c_relu'; /tmp/ccR5zelS.o:(.bss+0x10): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_core_layers.o):(.bss+0x38): multiple definition of `k2c_exponential'; /tmp/ccR5zelS.o:(.bss+0x8): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_core_layers.o):(.bss+0x40): multiple definition of `k2c_linear'; /tmp/ccR5zelS.o:(.bss+0x0): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_helper_functions.o):(.bss+0x0): multiple definition of `k2c_softsign'; /tmp/ccR5zelS.o:(.bss+0x40): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_helper_functions.o):(.bss+0x8): multiple definition of `k2c_softplus'; /tmp/ccR5zelS.o:(.bss+0x38): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_helper_functions.o):(.bss+0x10): multiple definition of `k2c_softmax'; /tmp/ccR5zelS.o:(.bss+0x30): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_helper_functions.o):(.bss+0x18): multiple definition of `k2c_sigmoid'; /tmp/ccR5zelS.o:(.bss+0x28): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_helper_functions.o):(.bss+0x20): multiple definition of `k2c_tanh'; /tmp/ccR5zelS.o:(.bss+0x20): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_helper_functions.o):(.bss+0x28): multiple definition of `k2c_hard_sigmoid'; /tmp/ccR5zelS.o:(.bss+0x18): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_helper_functions.o):(.bss+0x30): multiple definition of `k2c_relu'; /tmp/ccR5zelS.o:(.bss+0x10): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_helper_functions.o):(.bss+0x38): multiple definition of `k2c_exponential'; /tmp/ccR5zelS.o:(.bss+0x8): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_helper_functions.o):(.bss+0x40): multiple definition of `k2c_linear'; /tmp/ccR5zelS.o:(.bss+0x0): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_recurrent_layers.o):(.bss+0x0): multiple definition of `k2c_softsign'; /tmp/ccR5zelS.o:(.bss+0x40): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_recurrent_layers.o):(.bss+0x8): multiple definition of `k2c_softplus'; /tmp/ccR5zelS.o:(.bss+0x38): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_recurrent_layers.o):(.bss+0x10): multiple definition of `k2c_softmax'; /tmp/ccR5zelS.o:(.bss+0x30): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_recurrent_layers.o):(.bss+0x18): multiple definition of `k2c_sigmoid'; /tmp/ccR5zelS.o:(.bss+0x28): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_recurrent_layers.o):(.bss+0x20): multiple definition of `k2c_tanh'; /tmp/ccR5zelS.o:(.bss+0x20): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_recurrent_layers.o):(.bss+0x28): multiple definition of `k2c_hard_sigmoid'; /tmp/ccR5zelS.o:(.bss+0x18): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_recurrent_layers.o):(.bss+0x30): multiple definition of `k2c_relu'; /tmp/ccR5zelS.o:(.bss+0x10): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_recurrent_layers.o):(.bss+0x38): multiple definition of `k2c_exponential'; /tmp/ccR5zelS.o:(.bss+0x8): first defined here
/usr/bin/ld: ./include//libkeras2c.a(k2c_recurrent_layers.o):(.bss+0x40): multiple definition of `k2c_linear'; /tmp/ccR5zelS.o:(.bss+0x0): first defined here
collect2: error: ld returned 1 exit status
```

**Comment by roboticarm2000** [comment]
had the same issue.  The problem is that the declarations of k2c_activationType pointers are not declared extern in the header file.  Add the extern keyword, recompile the library, then the example.

extern k2c_activationType * k2c_linear;
extern k2c_activationType * k2c_exponential;
extern k2c_activationType * k2c_relu;
extern k2c_activationType * k2c_hard_sigmoid;
extern k2c_activationType * k2c_tanh;
extern k2c_activationType * k2c_sigmoid;
extern k2c_activationType * k2c_softmax;
extern k2c_activationType * k2c_softplus;
extern k2c_activationType * k2c_softsign;

**Comment by f0uriest** [comment]
Thanks for the fix @roboticarm2000 !
@rmcconke this should be fixed in #7
